File: 2 - Regression/5 - Multiple Linear Regression/multilpe_linear_regression.py
# -*- coding: utf-8 -*-

import numpy as np
import sklearn as sk
import pandas as pd
import matplotlib.pyplot as plt
import sklearn.model_selection._split
import sklearn.preprocessing as preprocess
import sklearn.linear_model

data = pd.read_csv('C:/DOC/Workspace/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 5 - Multiple Linear Regression/50_Startups.csv')
x, y = data.iloc[:, :-1].values, data.iloc[:, -1].values

# Categorical data
label_encoder = preprocess.LabelEncoder()
one_hot = preprocess.OneHotEncoder(categorical_features = [3])
x[:, 3] = label_encoder.fit_transform(x[:, 3])
x = one_hot.fit_transform(x).toarray()
# dummy variable trap
x = x[:, 1:]

# Test train split
x_train, x_test, y_train, y_test = sk.model_selection._split.train_test_split(x, y, train_size = 0.7)

# Scale data
scaler = preprocess.StandardScaler()
x_train[:, 2:] = scaler.fit_transform(x_train[:, 2:])    
x_test[:, 2:] = scaler.transform(x_test[:, 2:])

# Regression
regressor = sk.linear_model.LinearRegression()
regressor.fit(x_train, y_train)
pred_test = regressor.predict(x_test)

File: 2 - Regression/5 - Multiple Linear Regression/multilpe_linear_regression.R
data = read.csv('C:/DOC/Workspace/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 5 - Multiple Linear Regression/50_Startups.csv')

# Feature scaling
for (i in 1:3) {
  data[,i] = scale(data[, i])
}

# Categorical data
state_levels = levels(data$State)
data$State = factor(data$State, levels = state_levels, labels = seq(length(state_levels)))

# Test - train split
library('caTools')
is_train = sample.split(data$Profit, SplitRatio = 0.7)
data_train = subset(data, is_train)
data_test = subset(data, !is_train)

# Regression
regressor = lm(formula = Profit ~ ., data = data_train)
pred_test = predict(regressor, data_test)